src/s3fs.cpp

```diff
@@ -3368,10 +3368,17 @@ static int readdir_multi_head(const char* path, const S3ObjList& head, void* buf
 
         for(s3obj_list_t::iterator reiter = notfound_param.notfound_list.begin(); reiter != notfound_param.notfound_list.end(); ++reiter){
             int dir_result;
-            if(0 == (dir_result = directory_empty(reiter->c_str()))){
+            std::string reiter_str = *reiter;
+            if('/' != *reiter_str.begin()){
+                // path must start with '/'
+                reiter_str = "/" + reiter_str;
+                S3FS_PRN_INFO2("Applied prefix '/' to directory str %s", reiter_str.c_str());
```

Comment:
Two questions/comments here:

1. (comment) not sure this message is going to be useful to anyone but us while debugging the issue. If this is just a fix that needs to be applied by design then there isn't really much point in logging it (especially at info level)
2. Have we tested this fix mounting the entire bucket? Curious if `""` ends up being `"/"` and if that works the same way for ListBucket

Reply:
Removed the info and tested that mounting the entire bucket works